polaris.shopify.com/pages/api/v0/tokens/[tokens].tsx

```diff
@@ -97,67 +58,47 @@ const formatTokenGroup = (tokenGroup: TokenGroup, format: Format) => {
       .join('\n');
   }
 
-  return tokenGroup;
+  return tokenValues;
 };
 
 const handler = async (req: NextApiRequest, res: NextApiResponse) => {
   const formatParam = (req.query.format || 'json') as Format;
-  const schemeParam = (req.query.scheme || 'light') as Scheme;
+  const schemeParam = (req.query.scheme || 'light') as ColorScheme;
 
   if (typeof formatParam === 'string' && typeof schemeParam === 'string') {
-    const colorSchemeTokenGroup = colorSchemeMap[schemeParam];
-    const tokenGroupParam: any = req.query.tokens || '';
-    let tokenGroups: string[] = [];
+    const colorScheme = osColorSchemes[schemeParam] as ColorScheme;
+    const tokenGroupParam = (req.query.tokens || '') as TokenGroupKey;
+    let tokenData: TokenGroup = {};
 
     // Determine which list(s) we are querying for based on the token param
     if (tokenGroupParam === 'all') {
-      tokenGroups = tokenGroups.concat(staticTokenGroupKeys);
-    }
+      staticTokenGroupKeys.forEach((group) => {
+        const tokenGroup: TokenGroup = tokens[group] || {};
 
-    if (tokenGroupParam === 'all' || tokenGroupParam === 'colors') {
-      tokenGroups.push(colorSchemeTokenGroup);
+        tokenData = {...tokenData, ...tokenGroup};
+      });
     }
 
-    if (staticTokenGroupKeys.includes(tokenGroupParam)) {
-      tokenGroups.push(tokenGroupParam);
-    }
+    if (tokenGroupParam === 'all' || tokenGroupParam === 'colors') {
+      const colorSchemeTokenGroup: TokenGroup =
+        tokens.colorSchemes[colorScheme] || {};
```

Comment:
```suggestion
        tokens.colorSchemes[schemeParam] || {};
```
As mentioned above, we should use the `schemeParam` to access the `colorSchemes` token group.